### DISCUSSION TITLE
Using IBTrACS csv file format for Storm Dataset

### DISCUSSION THREAD

**Discussion by dooceye** [umr-amap/StormR#83] [Q&A]
I've been trying to use the .csv version of IBTrACS data as the dataset in the defStormsDataset() function, but I can't seem to read it in. Using the example code given in the [article](https://umr-amap.github.io/StormR/reference/defStormsDataset.html) seems to open up the .nc despite me having a .csv file with the same name in the same folder.

```
fileName <- system.file("extdata", "test_dataset.csv", package = "StormR")
SP_2015_2020_csv <- defStormsDataset(seasons = c(2010, 2021))

#> Warning: No basin argument specified. StormR will work as expected
#>              but cannot use basin filtering for speed-up when collecting data
#> === Loading data  ===
#> Open database... /home/runner/work/_temp/Library/StormR/extdata/test_dataset.nc opened
#> Collecting data ...
#> === DONE ===
```

If I remove the test_dataset.nc file from the folder, or try and get it to open a .csv which is stored in a different location I get the following error:

```
fileName <- system.file("extdata", "test_dataset.csv", package = "StormR")
SP_2015_2020_csv <- defStormsDataset(seasons = c(2010, 2021))
=== Loading data  ===
Open database... Error in ncdf4::nc_open(filename) : 
  Passed a filename that is NOT a string of characters!
In addition: Warning message:
In checkInputsdefStormsDataset(filename, fields, basin, seasons,  :
  No basin argument specified. StormR will work as expected
             but cannot use basin filtering for speed-up when collecting data
```

The csv I have been using is downloaded from: https://www.ncei.noaa.gov/data/international-best-track-archive-for-climate-stewardship-ibtracs/v04r00/access/csv/

Is there a different way of formatting the csv file that needs to happen before it can be recognised?

**Comment by thomasarsouze** [comment]
Hi @dooceye ,
sorry for this very late reply: somehow I didn't get any notification from your message.

This feature of using `.csv` dataset is quite new and so far only available on the github version. 
So you should install the latest version of the package using
```
devtools::install_github("umr-amap/StormR")
```

We will soon update the CRAN version so that it matches the documentation.
